### PR TITLE
feat: validate run tests

### LINKED ISSUE #782: Integrated Validation Testing
### Is your feature request related to a problem? Please describe.

Follow on to #460 - Identify the mechanism for integrating validation tests to `lula validate` and the expected/reported outcomes of the tests.

### Describe the solution you'd like

- **Given** Component definition with Lula Validations that have tests
- **When** `lula validate <> --run-tests`
- **Then** Reported test outcomes

### Expected Deliverable

- [ ] Validate Command updates -> Test Report
- [ ] Documentation with Test outputs for cmd

### PR DESCRIPTION
**Pull request by meganwolf0**
## Description

Instruments the test-report output for `lula validate ... --run-tests`
* Adds method to Validation Store to allow for execution of validation tests for all validations in the store
* Generates a test-report-{target}-{timestamp}.yaml with the content of the tests, written to the same directory as assessment results; includes summary of tests in console output of `lula validate...`
* Updated testing docs

**Notes to reviewers**
This PR has a lot of opinionation about how testing should be reported to the users of the functionality. Any and all feedback about what is useful/missing/ugly is much appreciated!

Additionally, the `lula dev validate` has an option to print-test-resources - that functionality was omitted here. Mostly because with many validations, each possibly with many tests, we're going to have a lot of output resources. Where should those be stored? Maybe in the `resources` directory, co-located with any `save-resouces` objects. Or in a separate `test-results` directory? But left it out for now, figured it'd be a quick add in the future if desired.

## Related Issue

Fixes #782

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [ ] (N/A)[Schema Updates](https://github.com/defenseunicorns/lula/blob/main/docs/community-and-contribution/schema-updates.md) applied
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/lula/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: docs/reference/testing.md [hunk 1]
```diff
@@ -128,17 +128,56 @@ Which will delete the existing labels map and then add an empty map, such that t
...
+          pass: true
+          result: not-satisfied
+        - test-name: no-containers
+          pass: true
+          result: not-satisfied
+```
+> Note that `61ec8808-f0f4-4b35-9a5b-4d7516053534` is the UUID of the validation without tests, and `82099492-0601-4287-a2d1-cc94c49dca9b` is the UUID of the validation with tests.
+
+### lula dev validate
+When executing `lula dev validate ... --run-tests`, the test results data will be written directly to console.
```
Comment (brandtkeller) [inline docs/reference/testing.md:164]: Documentation above mentions writing the results to a file (specifically markdown) whereas this references `directly to console` yet my local testing created a `.yaml` file.

**Review by brandtkeller — CHANGES_REQUESTED** [review]
Minor discrepancy in my review thus far.

File: docs/reference/testing.md (hunk 1 above)
Comment (meganwolf0) [reply docs/reference/testing.md:164]: ah sorry for the confusion - the behavior of `dev validate` is directly to console, `validate` prints a summary to console but the actual test results to a yaml doc. The md was an old thing and I'll update it! Mostly I just figured that printing test results to console for a single validation `dev validate` would be fine, but if you potentially had a ton you might want a separate output format?

Comment (brandtkeller) [reply docs/reference/testing.md:164]: appreciate the clarity and distinction between the commands. I think I got a little lost there and didn't notice the difference between the two commands.

**Review by brandtkeller — APPROVED** [review]

**Pull request merged**